### ISSUE TITLE
Improve guidelines for `sensor_location` attribute in hardware section

### ISSUE THREAD

**Issue by sebastien-rosset** [open-telemetry/opentelemetry-specification#3139]
**What are you trying to achieve?**

Improve semantic guidelines for `sensor_location` attribute.

**What did you expect to see?**

In the hardware semantic guidelines:

1. Provide optional attributes to link the temperature metrics to hardware components such as CPU, fan or battery.
1. The name of the `sensor_location` attribute should be hierarchical. See #3131.

**Additional context.**

Some components have a `sensor_location` attribute. For example, [fan](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/hardware-metrics.md#hwfan---fan-metrics), [temperature](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/hardware-metrics.md#hwtemperature---temperature-sensor-metrics) and [voltage](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/hardware-metrics.md#hwvoltage---voltage-sensor-metrics) have a `sensor_location` attribute.

The example values seem to match names found in IPMI.

It makes sense to use `sensor_location` for temperature and voltage, as this indicates the precise location of the sensor that measures the temperature metric or voltage metric. However, without additional attributes, it would be difficult to determine which physical component that temperature/voltage sensor is associated with. For example, if `sensor_location` is `CPU0_DIE`, I may be able to lookup a data sheet and find out it's the temperature sensor of the CPU in socket 0. But it wouldn't be easy to map it to a CPU slot or CPU vendor/model/serial.

It's odd to use `sensor_location` for a fan. A Fan is not just a sensor, it's a physical component that may have one or more sensors.

**Comment by bertysentry** [comment]
Unfortunately, it's difficult to mandate the proper identification of the location of a temperature sensor. Sometimes it's not even associated to another electronic component (like the sensors measuring the ambient temperature, in or out). 

But when the instrumentation layer can determine the component the sensor is attached to, then yes, as you suggested, the `parent` attribute should be used to properly associate a temperature metric (`hw.temperature{parent="cpu0"}`) with the corresponding device (when monitored with another metric, like `hw.status{id="cpu0"}`). To be even more precise, the parent type should be specified as well: `hw.temperature{parent_type="cpu", parent="cpu0"}`. But then, that make me think that the `parent` attributes should be properly namespaced:

* `hw.parent.id`
* `hw.parent.type`

What do you think?

WRT extra attributes like `vendor`, `model`, and `serial_number`, they are rarely available for temperature sensors, but the developer always has the ability to add such additional attributes, when they are relevant.

**Comment by sebastien-rosset** [comment]
> To be even more precise, the parent type should be specified as well: hw.temperature{parent_type="cpu", parent="cpu0"}

I agree, because it's possible that the same name may exist across two different parent types. This would help to disambiguate.

> But then, that make me think that the parent attributes should be properly namespaced

I'm not sure I'm following why adding a "parent_type" implies the attribute should be namespaced? Is it motivated by the fact there would be a new "hw.parent" namespace, and that namespace has at least two attributes (`id` and `type`)? If yes, I agree.

**Comment by ocelotl** [comment]
I asked Claude to review this issue and Claude suggests it be closed with this explanation:

> Purely about hardware-metrics semconv (sensor_location, parent linkage of temperature/voltage sensors to components). That doc moved to open-telemetry/semantic-conventions (model/hardware/), and an hw.parent attribute now exists there (common.yaml) partly addressing the parent-linkage proposal. Not actionable in the spec repo; belongs in semantic-conventions.

To guard against a mistaken suggestion, Claude then independently re-verified this conclusion:

> hardware-metrics.md is 404 in spec repo; content moved to semantic-conventions/model/hardware/. registry.yaml now defines hw.parent (parent linkage), hw.type enum (cpu/fan/temperature/voltage...), and hw.vendor/hw.model/hw.serial_number, plus namespaced hw.sensor_location - satisfying both of #3139's asks. Not actionable in spec repo.

Claude evaluated this issue with high confidence that it can be closed.

Claude has classified this issue as belonging in another repository.

The intention of this comment is to make it easier for a human being to review this issue before making a decision.

**Comment by svrnm** [comment]
Closing this, as this is now a semantic conventions issue, and requirements seem to be satisfied, if this is incorrect, please open a new issue on the semantic conventions repository.
